### PR TITLE
[CI] Run yamllint as an action to post.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Execute yamllint on the pipelines code and post a github formatted result in a comment.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

**Review by haritha-mohan — APPROVED** [review]

File: .github/workflows/yamllint.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# yamllint disable rule:line-length rule:document-start rule:truthy
+name: PR yamllint check
+on: pull_request
+
+# This action only need a single permission in order to autoformat the code.
```
Comment (rolfbjarne) [inline .github/workflows/yamllint.yml:5]: The action doesn't autoformat, so either the comment is wrong, or the code is.

File: .github/workflows/yamllint.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  rebase:
+    name: yamllint check
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0  # otherwise, you will fail to push refs to dest repo
```
Comment (rolfbjarne) [inline .github/workflows/yamllint.yml:20]: Same

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: .github/workflows/yamllint.yml (hunk 1 above)
Comment (mandel-macaque) [reply .github/workflows/yamllint.yml:5]: Happens when you yy a file to not start from scratch 😅

Comment (mandel-macaque) [inline .github/workflows/yamllint.yml:5]:
```suggestion
```

File: .github/workflows/yamllint.yml (hunk 2 above)
Comment (mandel-macaque) [inline .github/workflows/yamllint.yml:20]:
```suggestion
          fetch-depth: 0
```

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**